projects/platform/src/lib/pipes/call.pipe.ts

```diff
@@ -1,22 +1,25 @@
 import { ChangeDetectorRef, EmbeddedViewRef, Pipe, PipeTransform } from '@angular/core';
 
+type Fn = (...args: any[]) => any;
+type Head<T extends Fn> = Parameters<T>[0];
+type Tail<T extends Fn> = T extends (first: any, ...rest: infer R) => any ? R : never;
+type ClassMethod<C, K extends keyof C> = C[K] extends (...args: infer A) => infer R ? (...args: A) => R : never;
+
 @Pipe({ name: 'call' })
-export class CallPipe implements PipeTransform {
-  context: any;
+export class CallPipe<C> implements PipeTransform {
+  private readonly context: C;
 
-  constructor(cd: ChangeDetectorRef) {
-    this.context = (cd as EmbeddedViewRef<any>).context;
+  constructor(private cd: ChangeDetectorRef) {
+    this.context = (this.cd as EmbeddedViewRef<C>).context;
   }
 
-  transform(param: any, fn: string | Function, ...params: any[]) {
-    if (typeof fn === 'string') {
-      fn = this.context[fn];
-    }
-
-    if (typeof fn !== 'function') {
-      fn = () => { };
+  transform<M extends Fn>(param: Head<M>, fn: M, ...params: Tail<M>): ReturnType<M>;
+  transform<M extends ClassMethod<C, K>, K extends keyof C>(param: Head<M>, fn: K, ...params: Tail<M>): ReturnType<M>;
+  transform<M extends Fn | ClassMethod<C, K>, K extends keyof C>(param: Head<M>, fn: M | K, ...params: Tail<M>): ReturnType<M> {
+    if (typeof fn === 'function') {
+      return fn.apply(this.context, [param, ...params]);
     }
 
-    return fn.apply(this.context, [param, ...params]);
+    return (this.context[fn] as unknown as M).apply(this.context, [param, ...params]);
```

Comment:
Maybe using type guards will be better? What do you think?
```suggestion
    if (typeof fn === 'string') {
      fn = this.context[fn];
    }
    
    if (typeof fn === 'function') {
      return fn.apply(this.context, [param, ...params]);
    }
```

Reply:
do we need them? we have types in `transform` signature, it will show you an error if you try to pass something wrong